### PR TITLE
inital parsing of maintainers files

### PR DESCRIPTION
**Pull request by jfrazelle**
ping @thaJeztah @icecrime

### CODE REVIEW

**Comment by jfrazelle** [comment]
dont judge the code i wrote it fast :innocent:

**Comment by jfrazelle** [comment]
part of https://github.com/docker/docker/issues/18319

File: MAINTAINERS [hunk 1]
```diff
@@ -0,0 +1,169 @@
+[Projects]
+  [Projects.docker]
+    Url = "https://github.com/docker/docker"
```
Comment (icecrime) [inline MAINTAINERS]: Is `BurntSushi/toml` responsible for the weird indentation?

File: MAINTAINERS [hunk 2]
```diff
@@ -0,0 +1,169 @@
+[Projects]
```
Comment (icecrime) [inline MAINTAINERS]: It might need to be regenerated, I only see `docker` and `notary` here.

**Comment by icecrime** [comment]
That's really really cool, thanks @jfrazelle :-)

A few remarks:

- I think the [first ~200 lines](https://github.com/docker/docker/blob/bd2d53af2c6b8da6ab136298650318186cabbdf8/MAINTAINERS#L1-L202) of the `MAINTAINERS` files at `docker/docker` root should appear in this one. These are common rules and roles (like BDFL and chief architect).
- If we do so, than we should keep the lists being nested under `[Org]` (as in: `[Org.Maintainers.Docker]`, `[Org.Maintainers.Compose]`.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+.AppleDouble
+.LSOverride
+
+# Icon must ends with two \r.
+Icon
+
+
+# Thumbnails
+._*
+
```
Comment (thaJeztah) [inline .gitignore]: perhaps those `Thumb.db` files as well for windows users

File: MAINTAINERS (hunk 2 above)
Comment (icecrime) [inline MAINTAINERS]: Oh, actually that's probably because they aren't updated to TOML yet.

**Comment by thaJeztah** [comment]
Nice @jfrazelle!

obviously opinionated, but how do we handle the docs maintainers? They're in a separate list in the docker/docker maintainers file, but are maintainers :innocent: 

I can create pull requests for the other repos to convert them to TOML tomorrow

**Comment by jfrazelle** [comment]
oh this was my bad :) will update

**Comment by jfrazelle** [comment]
updated!

File: MAINTAINERS (hunk 1 above)
Comment (jfrazelle) [inline MAINTAINERS]: yes

File: MAINTAINERS (hunk 2 above)
Comment (jfrazelle) [inline MAINTAINERS]: true

**Comment by jfrazelle** [comment]
ah i see whill update per https://github.com/docker/opensource/pull/35#issuecomment-161787752

**Comment by jfrazelle** [comment]
so actually hmmmm thats hard because i cant get the comments re parsing, like should we just have that hardcoded?

**Comment by thaJeztah** [comment]
So, for the TOML, each repo should have a `[Org."Core Maintainers"]` entry, and that will be used?

**Comment by jfrazelle** [comment]
ya i made a pr here that is currently working https://github.com/docker/notary/pull/333

**Comment by thaJeztah** [comment]
Perhaps re: parsing, we should put those in a "description" node, or something?

**Comment by thaJeztah** [comment]
Ah cool, want me to do the other repos tomorrow, using the same format?

**Comment by jfrazelle** [comment]
ya so the rules can just live here and i will reformat the other roles to not be comments in this pr

**Comment by thaJeztah** [comment]
Cool, thanks Jess!

**Comment by jfrazelle** [comment]
ok so updated

**Comment by jfrazelle** [comment]
im going to merge this just so i can set up automation to cron job and update :innocent:

**Pull request merged**